### PR TITLE
GPII-3324: Add gcp-stackdriver-export

### PR DESCRIPTION
**Pull request by mrtyler**
This PR is on top of my branch for #114.

Requires https://github.com/gpii-ops/exekube/pull/16.

Next steps:
* Retain some container logs. Only the gpii namespace? Some other set?

### CODE REVIEW

File: gcp/modules/gcp-stackdriver-export/main.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    # TODO: This is a lot less noisy if we add "AND NOT operation.producer=k8s.io".
+    "cloudaudit-activity" = "logName=projects/__var.project_id__/logs/cloudaudit.googleapis.com%2Factivity"
+
+    # Captures Google Storage and KMS events
+    "cloudaudit-data-access" = "logName=projects/__var.project_id__/logs/cloudaudit.googleapis.com%2Fdata_access"
+
+    # Captures events for products within GCE, such as Snapshots, Instance
+    # Groups, Firewalls, etc.
+    "compute-activity" = "logName=projects/__var.project_id__/logs/compute.googleapis.com%2Factivity_log"
+  }
```
Comment (stepanstipl) [inline gcp/modules/gcp-stackdriver-export/main.tf]: Seems like we have the `${var.project_id}` var here, is there a `__var.project_id__` need for the replacement bit?

Comment (mrtyler) [reply gcp/modules/gcp-stackdriver-export/main.tf]: This discussion is continuing in https://github.com/gpii-ops/exekube/pull/12.

**Comment by mrtyler** [comment]
> do we want to export only a subset of the logs? (as opposed to exporting all the logs from the project)

I think yes.

We discussed this a bit in https://issues.gpii.net/browse/GPII-3324. The idea is to differentiate logs that may be useful in the long-term (like audit logs) from (generally noisy) logs that are unlikely to be useful in the long-term (like HTTP access logs).

A related problem: I haven't tested this yet but it looks like exports ignore exclusion rules:
https://cloud.google.com/logging/docs/exclusions#overview

I don't think we want to do long-term archiving of Calico and Kubelet noise that we are excluding from Stackdriver. I'm not sure how to filter the noise if we choose to export all logs (possibly sending noisy logs to a different GS bucket with a different lifecycle policy).

What do you think?

**Comment by natarajaya** [comment]
LGTM.

+1 to export only logs that can be useful in long-term.
Regarding `__var.project_id__` substitution, please check [my comment](https://github.com/gpii-ops/gpii-infra/pull/114#issuecomment-425268582) under  https://github.com/gpii-ops/gpii-infra/pull/114.

**Comment by stepanstipl** [comment]
Ok, as long as we're careful to not leave out anything important, I'm happy.  👍 

Btw. it's not very nice you can't use same exclusions you do for dashboard in exports :(.

**Comment by mrtyler** [comment]
WTF github. I did not close this PR. The "reopen" button is grayed out because "The gpii-3327-exclude branch has been deleted" (it hasn't).

**Comment by mrtyler** [comment]
Continued in https://github.com/gpii-ops/gpii-infra/pull/156.

Sorry about this. I prefer to keep all the discussion in one place but I can't convince Github to re-open this PR :/.

**Pull request closed without merging**